python/ray/reporter.py

```diff
@@ -107,6 +113,27 @@ def __init__(self, redis_address, redis_password=None):
     def get_cpu_percent():
         return psutil.cpu_percent()
 
+    @staticmethod
+    def get_gpu_usage():
+        if gpustat is None:
+            return []
+        gpu_utilizations = []
+        gpus = []
+        try:
+            gpus = gpustat.new_query().gpus
```

Comment:
not sure how expensive this call is if there are no gpus - can we avoid calling it when we know there aren't any? maybe check if gpus were specified to ray on the node?

Reply:
https://github.com/wookayin/gpustat/blob/master/gpustat/core.py
Looking here it seems to me the call would be somewhat cheap. It makes a sys call to get the number of GPUs before doing anything. 

What about an approach where if it detects no GPUs for a certain period of time it logs a debug message and stops polling? I think that might be easier than working in the Ray cluster data.

Reply:
Let's just leave it in for now, that seems complicated. I would just leave a TODO as a possible optimization.